fix: use the updated link checker fork

Skarlso opened the pull request:
<!-- markdownlint-disable MD041 -->
#### What this PR does / why we need it

#### Which issue(s) this PR fixes
<!--
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
-->

frewilhelm commented:
This will fix https://github.com/open-component-model/ocm/issues/1474

frewilhelm commented:
Did you had any chance to check if this works?

frewilhelm commented:
Yes, I was wondering about the same, when I took a look yesterday. Apparently, it was working fine until the update of the original action.

Skarlso commented:
Okay, the check is now fixed. Funny enough `,{ "pattern": "^{{.*}}$" }` we had in there actually doesn't work, because it's being translated into the regex I added now.

Skarlso commented on .github/config/markdown-link-check.json:
```diff
@@ -5,6 +5,7 @@
     ,{ "pattern": "^{{.*}}$" }
-    ,{ "pattern": "^https://github.com/**/compare/**" }
+    ,{ "pattern": "^https://github.com/.*/compare/.*" }
+    ,{ "pattern": "%7B%7B.*%7D%7D" }
```
let it be known now and for future reference.

The above `"^{{.*}}$"` is actually not working, because the link being checked is URL encoded BEFORE checking. Meaning the actual pattern of the link that we'd need to ignore in case of our lovely templated links is this one.

Skarlso commented:
Previous failed run: https://github.com/open-component-model/ocm/actions/runs/15159082663/job/42620755562
Previous passed run when I edited a markdown: https://github.com/open-component-model/ocm/actions/runs/15159137065
